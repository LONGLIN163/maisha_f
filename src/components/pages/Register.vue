<template>
    <div>
       <van-nav-bar
        title="Sign Up"
        left-text="back"
        left-arrow
        @click-left="goBack"
        />

        <div class="register-panel">
        <van-field
            v-model="username"
            label="User Name"
            icon="clear"
            placeholder="Please input user name"
            required
            @click-icon="username = ''"
        />

        <van-field
            v-model="password"
            type="password"
            label="Password"
            placeholder="Please input password"
            required
        />
        <div class="register-button">
            <van-button type="primary" @click="register" size="large">Register Now</van-button>
        </div>
       </div>

    </div>
</template>

<script>
    import axios from 'axios'
    import serviceApi from '@/serviceApi.config.js'

    export default {
        data() {
            return {
                username: '',
                password: '',
            }
        },
        methods: {
            goBack() {
                this.$router.go(-1)   
            },
            register(){
                alert('register')
                axios({
                    url:serviceApi.userRegister,
                    method:'POST',
                    data:{
                       username:this.username,
                       password:this.password
                    }
                })
                .then(response => {
                    console.log(response)
                })
                .catch((error) => {
                    console.log(error)
                })
            }
        },
    }
</script>

<style scoped>
    .register-panel{
        width:96%;
        border-radius: 5px;
        margin:20px auto;
        padding-bottom:50px;
    }
    .register-button{
        padding-top:10px;
    }
</style>