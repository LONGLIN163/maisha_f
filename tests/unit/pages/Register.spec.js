
import { mount, createLocalVue} from '@vue/test-utils'
import Register from '@/components/pages/Register.vue'

import VueRouter from 'vue-router';
const Vant=require("vant")

const localVue = createLocalVue();
localVue.use(VueRouter); 
localVue.use(Vant); 

const router = new VueRouter();

describe('Register',  () => {

  let wrapper;
  // const goBack = jest.spyOn(Register.methods, 'goBack')
  //   .mockImplementation(() => {
  //     console.log("1. mock goBack implementation")
  //   })

  // mock loginAction()
  const registerAction = jest.spyOn(Register.methods, 'registerAction')
    .mockImplementation(() => {
      //console.log("1. mock registerAction implementation")
      checkForm() && register() // call checkForm(set true), then call login
  })

  // mock checkForm()
  const checkForm = jest.spyOn(Register.methods, 'checkForm')
    .mockImplementation(() => {
      //console.log("2. mock checkForm implementation")
      return true
    })

  // mock register()
  const register = jest.spyOn(Register.methods, 'register')
    .mockImplementation(() => {
      //console.log("3. mock register implementation")
  })

  beforeEach(() => {
    wrapper = mount(Register,{
      localVue,
      router
    })
  })

  it('renders Register',async () => {
    expect(wrapper).toMatchSnapshot()
  })  

  // ***this would not work, jest would not trigger "click-left" event by "click", this has to be done by e2e
  // it('goBack() should have been called correctly ',async () => {
  //   console.log(wrapper.find('[data-test="goBack"]').text())
  //   await wrapper.find('[data-test="goBack"]').trigger('click');
  //   expect(goBack).toHaveBeenCalled()
  // })

  it('test register input field ',async () => {
    await wrapper.find('[data-test="username"]').setValue("usr");
    expect(wrapper.find('[data-test="username"]').element.value).toBe('usr')

    await wrapper.find('[data-test="password"]').setValue("pwd");
    expect(wrapper.find('[data-test="password"]').element.value).toBe('pwd')
  })

  it('registerAction() should have been called correctly ',async () => {
    await wrapper.find('[data-test="registerAction"]').trigger('click');
    expect(registerAction).toHaveBeenCalled()
  })

})