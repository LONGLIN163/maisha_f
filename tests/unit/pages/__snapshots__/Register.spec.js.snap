// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders Register 1`] = `
<div>
  <div class="van-nav-bar van-hairline--bottom" data-test="goBack">
    <div class="van-nav-bar__content">
      <div class="van-nav-bar__left"><i class="van-icon van-icon-arrow-left van-nav-bar__arrow">
          <!---->
        </i><span class="van-nav-bar__text">back</span></div>
      <div class="van-nav-bar__title van-ellipsis">Sign Up</div>
    </div>
  </div>
  <div class="register-panel">
    <div class="van-cell van-cell--required van-field">
      <div class="van-cell__title van-field__label"><span>User Name</span></div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body"><input type="text" data-test="username" placeholder="Please input user name" class="van-field__control"></div>
      </div>
    </div>
    <div class="van-cell van-cell--required van-field">
      <div class="van-cell__title van-field__label"><span>Password</span></div>
      <div class="van-cell__value van-field__value">
        <div class="van-field__body"><input type="password" data-test="password" placeholder="Please input password" class="van-field__control"></div>
      </div>
    </div>
    <div class="register-button"><button data-test="registerAction" class="van-button van-button--primary van-button--large">
        <div class="van-button__content"><span class="van-button__text">Register Now</span></div>
      </button></div>
  </div>
</div>
`;
